python/lsst/ip/isr/isrFunctions.py

```diff
@@ -546,10 +560,30 @@ def overscanCorrection(ampMaskedImage, overscanImage, fitType='MEDIAN', order=1,
         stdevBiasArr = 0.74*(percentiles[2] - percentiles[0])  # robust stdev
         diff = numpy.abs(biasArray - medianBiasArr[:, numpy.newaxis])
         biasMaskedArr = numpy.ma.masked_where(diff > numSigmaClip*stdevBiasArr[:, numpy.newaxis], biasArray)
-        collapsed = numpy.mean(biasMaskedArr, axis=1)
-        if collapsed.mask.sum() > 0:
-            collapsed.data[collapsed.mask] = numpy.mean(biasArray.data[collapsed.mask], axis=1)
+
+        if fitType in ('MEDIAN_PER_ROW',):
+            collapsed = []
+            fitTypeStats = afwMath.stringToStatisticsProperty('MEDIAN')
+            # Loop over rows because there is no afwMath.Median(array, axis=1)
+            for row in biasMaskedArr:
+                if row.mask is False:
+                    row.mask = numpy.repeat(False, len(row))
+                medianRow = afwMath.makeStatistics(row, row.mask, fitTypeStats, statControl).getValue()
+                collapsed.append(medianRow)
+            collapsed = numpy.array(collapsed)
+            # Still need mask: use numpy to obtain the collapsed mask from biasMaskedArr
+            a = numpy.median(biasMaskedArr, axis=1)
+            collapsed = numpy.ma.masked_array(collapsed, mask=a.mask)
+            fitBiasArr = collapsed
```

Comment:
I'm not quite sure why this whole block doesn't like in the big block above, but maybe I'm missing something.

Reply:
EDIT: I tried moving it and it was a bad bad time. It should live here, and this code *really* needs a refactor!

Reply:
At first, I thought about including the new code where you say, but then I saw that the other part (the part that collapses the overscan for the polynomial corrections) had stuff that included dealing with long vs short axis, clipping, and masking, so I thought we wanted to include that. I had to fiddle with the code to make it pass the tests (and probably made it uglier by adding more control statements; I'm sorry). However---and as you have corroborated independently too---when you move things around then you end up breaking other parts.

Reply:
If you want, I could try to move the loop to that other part and see if I can make things work.

Reply:
Yeah, this whole function is incredibly unwieldy and in desperate need of refactoring. Let's leave this here for now (as discussed OOB).